frontend/public/components/debug-terminal.tsx

```diff
@@ -0,0 +1,226 @@
+import { PodKind, k8sCreate, k8sKillByName } from '@console/internal/module/k8s';
+import { PodExecLoader } from '@console/internal/components/pod';
+import { PodModel } from '@console/internal/models';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+
+import { resourcePath } from './utils/resource-link';
+import { getBreadcrumbPath } from '@console/internal/components/utils/breadcrumbs';
+import { isWindowsPod } from '../module/k8s/pods';
+
+const getDebugPod = (debugPodName: string, podToDebug: PodKind, containerName: string) => {
+  const debugPod: PodKind = _.cloneDeep(podToDebug);
+  delete debugPod.metadata.resourceVersion;
+  delete debugPod.metadata.uid;
+  delete debugPod.metadata.managedFields;
+  delete debugPod.metadata.name;
+  debugPod.metadata.generateName = debugPodName;
+  debugPod.metadata.annotations['debug.openshift.io/source-container'] = containerName;
+  debugPod.metadata.annotations[
+    'debug.openshift.io/source-resource'
+  ] = `/v1, Resource=pods/${podToDebug?.metadata?.name}`;
+  debugPod.spec.restartPolicy = 'Never';
+
+  debugPod.spec.containers.forEach((container) => {
+    container.command = isWindowsPod(podToDebug) ? ['cmd'] : ['/bin/sh'];
+    container.stdin = true;
+    container.stdinOnce = true;
+    container.tty = true;
+    delete container?.readinessProbe;
+    delete container?.livenessProbe;
+  });
+  delete debugPod.spec.host;
+  delete debugPod.spec.nodeName;
+  delete debugPod.status;
+
+  return debugPod;
+};
+
+const DebugTerminalError: React.FC<DebugTerminalErrorProps> = ({ error, description }) => {
+  return (
+    <div className="co-m-pane__body">
+      <Alert variant="danger" isInline title={error}>
+        <p>{description}</p>
+      </Alert>
+    </div>
+  );
+};
+
+const DebugTerminalInner: React.FC<DebugTerminalInnerProps> = ({ debugPod, initialContainer }) => {
+  const { t } = useTranslation();
+  const infoMessage = (
+    <div className="co-terminal-info-message">
+      <Alert
+        variant="info"
+        isInline
+        title={t(
+          'public~This temporary pod has a modified entrypoint command to debug a failing container. The pod will be deleted when the terminal window is closed.',
+        )}
+      />
+    </div>
+  );
+  switch (debugPod?.status?.phase) {
+    case 'Failed':
+    case 'Unknown':
+      return (
+        <DebugTerminalError
+          error={<>{t('public~The debug pod failed. ')}</>}
+          description={
+            debugPod.status.containerStatuses?.[0]?.state?.terminated?.message ||
+            debugPod.status.message
+          }
+        />
+      );
+    case 'Running':
+      return (
+        <PodExecLoader
+          obj={debugPod}
+          initialContainer={initialContainer}
+          infoMessage={infoMessage}
+        />
+      );
+    case 'Pending':
+      return (
+        <LoadingBox message={debugPod.status.phase}>{debugPod.status?.message || ''}</LoadingBox>
```

Comment:
Yeah, this is what I was thinking. It is probably better to use 

```ts
debugPod.status?.containerStatuses?.[0]?.state?.waiting?.message || debugPod.status?.message
```

Reply:
Changed the message and squashed the commits